Bootstrap hugo/docsy site, move documentation

asmacdo opened the pull request:
### This branch is hosted at http://operator-sdk.netlify.com

Original proposal https://github.com/operator-framework/operator-sdk/blob/eb270396c47f3b9abedb34db2aed964a8a91f74e/doc/proposals/hugo-doc-build.md

Bootstrap hugo/docsy site, move documentation
 - Used `git mv` where possible
 - Each documentation file now includes Hugo friendly frontmatter.
 - Implemented a very simple landing page
 - Added a section for how to build these docs locally.
 - Update gitignore

https://github.com/operator-framework/operator-sdk/issues/2142
Fixes #2142

The content in docs index and the contributor docs index, was moved from README.md and 
CONTRIBUTING.md. `git mv` was used with those files, but overwritten when the minimal content that remains was added. The "new" content is just moved, and any content changes that need to be made should be addressed in a separate PR.

camilamacedo86 commented on README.md:
```diff
@@ -6,189 +6,25 @@
+[controller-runtime][controller_runtime] library to make writing
+operators easier by providing:
+- High level APIs and abstractions to write the operational logic more
+- intuitively
```
I think that makes sense keep here just the overview info but not change it. 
Also, we should add the link for the web docs as well.  Which in POV means that it should be done just when the web docs be 100% ok and not in this PR.

camilamacedo86 commented on README.md:
```diff
@@ -6,189 +6,25 @@
-- Access to a Kubernetes v1.11.3+ cluster.
-- Optional: [delve](https://github.com/go-delve/delve/tree/master/Documentation/installation) version 1.2.0+ (for `up local --enable-delve`).
-
-## Quick Start
```
IHMO the Quick Start should be an doc just for it as it is for kubebuilder.

camilamacedo86 commented:
Hi @asmacdo, 

It is missing fix the broken links to pass in the CI. (marker test).
Also, following my review:

- Should it really be operator-sdk.netlify.com? Why netlify? Could we not follow the same k8s standard and be https://operator-sdk.io?
- The home page and icon should be : https://github.com/operator-framework/operator-sdk/blob/master/doc/images/operator_logo_sdk_color.svg
- See my comment [here](https://github.com/operator-framework/operator-sdk/pull/2155/files#r342679695). If we move with this PR we will not have it done and the users will no longer have the current info as well. So, we need a better approach to be able to work with follow up pr's until the web doc be 100% whiteout broke what is currently working.
- Regards the layout: I think we should use white and red instead of blue. 
- The search tool is not working and why have 2 inputs for it?
- I am not happy the current navigation, but I understand that it would be a follow-up PR. Where are tracked all steps/tasks that should be done until it is 100% completed? 
- The contribution guideline is not working. See https://operator-sdk.netlify.com/docs/contribution-guidelines/ ( this link is in the community section ) 
- Why we are adding a link for the stackoverflow? We should redirect the users to raise and check the issues in the repo.

jmrodri commented on .gitmodules:
```diff
@@ -0,0 +1,3 @@
+[submodule "website/themes/docsy"]
+	path = website/themes/docsy
+	url = https://github.com/google/docsy.git
```
I know the hugo examples all use submodules for the themes. But they're really annoying. We can try it but I'd almost rather just copy in the theme and deal with updates by hand if we ever need to update it from the upstream.

asmacdo replied:
I share your irritation with submodules and have been considering alternatives, but I haven't found a better solution. The biggest concern is that the docsy theme has a lot of files, and I don't think we want to carry something like 12k lines of javascript.

jmrodri commented:
Marker is failing on a bunch of stuff that isn't our docs:
```
ts/vendor/bootstrap/.github/CONTRIBUTING.md:147 : 404 Not Found
Found broken url       (searched for existing issues -> https://github.com/FortAwesome/Font-Awesome-Pro/issues) at ./website/themes/docsy/assets/vendor/Font-Awesome/.github/ISSUE_TEMPLATE/feature_request.md:25 : 404 Not Found
Found broken url       (5.0.0-alpha1 -> https://github.com/FortAwesome/Font-Awesome-Pro/releases/tag/5.0.0-alpha1) at ./website/themes/docsy/assets/vendor/Font-Awesome/CHANGELOG.md:872 : 404 Not Found
Found broken url       (Each test should begin with  -> https://api.qunitjs.com/assert/expect/) at ./website/themes/docsy/assets/vendor/bootstrap/js/tests/README.md:27 : 404 Not Found
Found broken url       (5.0.0-beta6 -> https://github.com/FortAwesome/Font-Awesome-Pro/releases/tag/5.0.0-beta6) at ./website/themes/docsy/assets/vendor/Font-Awesome/CHANGELOG.md:682 : 404 Not Found
Found broken url       (https://testydocsy.netlify.com/ -> https://testydocsy.netlify.com/) at ./website/themes/docsy/userguide/README.md:3 : 404 Not Found
Found broken url       (5.0.0-alpha5 -> https://github.com/FortAwesome/Font-Awesome-Pro/releases/tag/5.0.0-alpha5) at ./website/themes/docsy/assets/vendor/Font-Awesome/CHANGELOG.md:812 : 404 Not Found
Found broken url       (Hugo theme -> https://gohugo.io/themes/installing-and-using-themes/) at ./website/themes/docsy/userguide/README.md:1 : 404 Not Found
Found broken url       (Install and Use Themes -> https://gohugo.io/themes/installing-and-using-themes/#install-a-single-theme) at ./website/themes/docsy/userguide/content/en/docs/Getting started/_index.md:206 : 404 Not Found
Found broken url       (5.0.0-alpha2 -> https://github.com/FortAwesome/Font-Awesome-Pro/releases/tag/5.0.0-alpha2) at ./website/themes/docsy/assets/vendor/Font-Awesome/CHANGELOG.md:853 : 404 Not Found
Found broken url       (5.0.0-beta4 -> https://github.com/FortAwesome/Font-Awesome-Pro/releases/tag/5.0.0-beta4) at ./website/themes/docsy/assets/vendor/Font-Awesome/CHANGELOG.md:713 : 404 Not Found
make: *** [test/markdown] Error 1
```
We should probably update the `Makefile` to exclude those from the marker target

asmacdo commented:
@camilamacedo86 

> Should it really be operator-sdk.netlify.com? Why netlify? Could we not follow the same k8s standard and be https://operator-sdk.io

This is just a temporary namespace, which we can change in the netlify configuration whenever we register a domain. +1 operator-sdk.io

> The home page and icon should be: https://github.com/operator-framework/operator-sdk/blob/master/doc/images/operator_logo_sdk_color.svg

I experimented with this briefly, but the logo with words didn't fit the default templates. If we have a logo without the words, that should will work.

> Regards the layout: I think we should use white and red instead of blue. So, my suggestion would be to replace the blue for white or black(because of the sdk log is red), and the grey area for red.

These are the default colors, they can be changed very easily in a separate story.

> The search tool is not working and why have 2 inputs for it?

Search will be disabled in the next push and I will file a separate issue-- there is some extra config that needs to be done for search to work. We should be able to disable one of the search inputs.

> I am not happy the current navigation, but I understand that it would be a follow-up PR. Where are tracked all steps/tasks that should be done until it is 100% completed?

Organization issue is here. https://github.com/operator-framework/operator-sdk/issues/2143
Part of that issue will be to file more issues for new pages if necessary, which will also be filed here in GitHub.

> The contribution guideline is not working. See https://operator-sdk.netlify.com/docs/contribution-guidelines/ ( this link is in the community section )

Thank you! Fixed in the next push.

> Why we are adding a link for the StackOverflow? We should redirect the users to raise and check the issues in the repo.

I assumed that since we already have a tag, StackOverflow would be a better place for support than our issue tracker. I don't mind removing it if you feel strongly.

> Regards the navigation bar, I think we should add first Getting started, then the others as it is.

I added your suggestion to https://github.com/operator-framework/operator-sdk/issues/2143

camilamacedo86 commented:
Hi @jmrodri and @asmacdo, 

My big thing to merge this PR is because it is changing the current docs. The README for example and we have not the web docs done 100% after that.

Also,  regards the Maker test note that is passing in the other PR"s and the issue is related to the bootstrap added here `ts/vendor/bootstrap/.github/CONTRIBUTING.md:147 : 404 Not Found`. So, it needs to be fixed fore we are able to merge this one.

camilamacedo86 commented:
Regards the comment https://github.com/operator-framework/operator-sdk/pull/2155#issuecomment-549968920. 

I think we need to track all steps required from end to end regards to it. So, my suggestion would be created an issue with the TODO list as the action plan to let us know what is missing to do.

- [ ] redirect netlify donmain to operator-sdk.io: `This is just a temporary namespace, which we can change in the netlify configuration whenever we register a domain. +1 operator-sdk.io`
- [ ] Use the operator SDK logo (Note that we can just cut the image from the text if it be required) 
- [ ] Adjust the layout (My suggestion would be using the product colours instead of blue, which would replace the blue for white or black, because of the SDK log is red, and the grey area for red. )
- [ ]  Make the search work (related to `Search will be disabled in the next push and I will file a separate issue-- there is some extra config that needs to be done for search to work. We should be able to disable one of the search inputs.`)
- [ ] Organize docs : https://github.com/operator-framework/operator-sdk/issues/2143
- [ ] Remove StackOverflow links and references. Use the links and refs that are described in the docs to raise issues and etc ...

Also, add the link for the proposal. WDYT?

camilamacedo86 commented on website/content/en/docs/_index.md:
```diff
@@ -0,0 +1,209 @@
+[controller-runtime][controller_runtime] library to make writing
+operators easier by providing:
+- High level APIs and abstractions to write the operational logic more
+- intuitively
```
```suggestion
```

camilamacedo86 replied:
It is not in the original text I think

asmacdo commented on website/content/en/docs/_index.md:
```diff
@@ -0,0 +1,208 @@
+[operator_link]: https://coreos.com/operators/
+[proposals_docs]: /docs/proposals
+[sdk_cli_ref]: {{< ref "sdk-cli-reference.md" >}}
+[guide]: /docs/user/
```
This link fails the `marker` tests. Alternative syntax is relative, but AFAICT we cannot make `marker` play nice with Hugo. `marker` checks paths to files, but Hugo converts the markdown to html, and the paths to not resolve to files.

To move forward with Hugo, we will need to remove the `marker` tests. If we want to keep link checking, we should be able to add an html checker, and run against built docs. Anyone have a recommendation for such a tool?

joelanford replied:
Would something like this work: https://github.com/raviqqe/muffet?

camilamacedo86 commented on website/config.toml:
```diff
@@ -0,0 +1,127 @@
+# Everything below this are Site Params
+
+[params]
+copyright = "The Operator-SDK Authors"
```
```suggestion
copyright = "Operator SDK"
```

camilamacedo86 commented on website/content/en/docs/contribution-guidelines/docs/select-syntax.md:
```diff
@@ -0,0 +1,35 @@
+---
+title: Selected Syntax
```
```suggestion
title: Web doc Syntax
```
Something as ..

camilamacedo86 commented on website/content/en/docs/design/long-term/norms-to-follow.md:
```diff
@@ -1,4 +1,7 @@
-## SDK examples/docs show API norms authors should follow
+---
+title: SDK examples/docs show API norms authors should follow
```
```suggestion
title: SDK examples/docs and API norms
```

camilamacedo86 commented on website/content/en/docs/migration/version-upgrade-guide.md:
```diff
@@ -1,4 +1,7 @@
-# Operator SDK Version upgrade guide
+---
+title: Operator SDK Version upgrade guide
```
```suggestion
title: Version upgrade guide from v0.2.x to latest version
```

camilamacedo86 commented on website/content/en/docs/project_layout.md:
```diff
@@ -1,4 +1,8 @@
-# Project Scaffolding Layout for Operator SDK
+---
+title: Project Scaffolding Layout for Operator SDK
+linkTitle: Project Layout
```
```suggestion
linkTitle: Project Scaffolding Layout
```

camilamacedo86 commented on website/config.toml:
```suggestion

[[params.links.user]]
 	name = "Repo Issues"
 	url = "https://github.com/operator-framework/operator-sdk/issues/new/choose"
 	icon = "fab fa-github"
         desc = "Raise issues to get support, track bugs and request new features"
Developer relevant links. These will show up on right side of footer and in the community page if you have one.


```
